plugin/src/main/java/org/opennms/velocloud/requisition/PartnerRequisitionProvider.java

```diff
@@ -62,6 +70,63 @@ protected Requisition handleRequest(final Request request, final VelocloudApiCli
         final var requisition = ImmutableRequisition.newBuilder()
                 .setForeignSource(request.getForeignSource());
 
+        try {
+            for (var enterprise : client.getEnterprises()) {
+                final List<User> users = client.getUsers(enterprise.id);
+
+                //TODO: confirm  gateway roles are coming from enterprise users
+                //TODO: confirm Gateway AssetNodes destination.
+                client.getGateways(enterprise.enterpriseId).stream().forEach(gateway -> {
+                    requisition.addNode(ImmutableRequisitionNode.newBuilder()
+                            .setForeignId(gateway.gatewayId)
+                            .setNodeLabel(gateway.gatewayName)
+                            .setLocation(gateway.siteName)
+                            .setInterfaces(List.of(ImmutableRequisitionInterface.newBuilder()
+                                    .setIpAddress(gateway.ipAddress)
+                                    .build()))
+                            .addMetaData(ImmutableRequisitionMetaData.newBuilder()
+                                    .setContext(VELOCLOUD_CONTEXT)
+                                    .setKey("roles")
+                                    .setValue(users.stream().map(u ->
+                                            String.format("%s:%s", u.roleId, u.roleName)).collect(Collectors.joining(",")))
```

Comment:
```suggestion
                                            String.format("%s:%s", u.roleId, u.roleName)).collect(Collectors.joining("\n")))
```

we can have multi-line values there.